pkg/transforms/encryption.go

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright (c) 2017 Cavium
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package transforms
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/appcontext"
+)
+
+type Encryption struct {
+	Key string
+	Iv  string
+}
+
+// IV and KEY must be 16 bytes
+const blockSize = 16
+
+func pkcs5Padding(ciphertext []byte, blockSize int) []byte {
+	padding := blockSize - len(ciphertext)%blockSize
+	padtext := bytes.Repeat([]byte{byte(padding)}, padding)
+	return append(ciphertext, padtext...)
+}
+
+func (aesData Encryption) AESTransform(edgexcontext *appcontext.Context, params ...interface{}) (bool, interface{}) {
+	var data []byte
+	var err error
+	switch params[0].(type) {
```

Comment:
Time to create a helper func for this? How about a file in transforms called inputdata.go with a func called getInputData()

Reply:
i was thinking a function called normalizeInput(), but yeah lets put that in other pull request as a refactor for all the functions